### PR TITLE
chore: support async in rest call common macro

### PR DESCRIPTION
**Pull request by ohmayr**
This PR updates the `rest_call_method_common` macro to be re-used for asynchronous calls.

### CODE REVIEW

File: gapic/templates/%namespace/%name_%version/%sub/services/%service/_shared_macros.j2
Comment (parthea) [inline gapic/templates/%namespace/%name_%version/%sub/services/%service/_shared_macros.j2:142]: Please could you add a comment to explain the parameters of the macro?

File: gapic/templates/%namespace/%name_%version/%sub/services/%service/_shared_macros.j2
Comment (parthea) [inline gapic/templates/%namespace/%name_%version/%sub/services/%service/_shared_macros.j2:144]: I couldn't determine if this is used. Remove if it's not needed.

File: gapic/templates/%namespace/%name_%version/%sub/services/%service/_shared_macros.j2
Comment (parthea) [inline gapic/templates/%namespace/%name_%version/%sub/services/%service/_shared_macros.j2:148]: Add a todo comment with a link to a bug if this is a feature gap that needs to be fixed later

File: gapic/templates/%namespace/%name_%version/%sub/services/%service/_shared_macros.j2
Comment (parthea) [inline gapic/templates/%namespace/%name_%version/%sub/services/%service/_shared_macros.j2:166]: Add a comment to clarify the reason that `format_http_response_error` has more parameters for async compared to sync. If the sync code will be cleaned up later, add a TODO comment with a link to a bug

**Review by parthea — APPROVED** [review]

File: gapic/templates/%namespace/%name_%version/%sub/services/%service/_shared_macros.j2
Comment (parthea) [reply gapic/templates/%namespace/%name_%version/%sub/services/%service/_shared_macros.j2:144]: If `async_prefix = "async "` is unused, I would recommend to remove it. I couldn't find `async_prefix` in the macro other than the location where it is defined

Comment (ohmayr) [reply gapic/templates/%namespace/%name_%version/%sub/services/%service/_shared_macros.j2]: Apologies, I missed that! You're right and that's a good catch. I've removed it now!

File: gapic/templates/%namespace/%name_%version/%sub/services/%service/_shared_macros.j2
Comment (ohmayr) [reply gapic/templates/%namespace/%name_%version/%sub/services/%service/_shared_macros.j2:155]: Addressed.

File: gapic/templates/%namespace/%name_%version/%sub/services/%service/_shared_macros.j2
Comment (ohmayr) [reply gapic/templates/%namespace/%name_%version/%sub/services/%service/_shared_macros.j2:178]: Addressed.

**Comment by ohmayr** [comment]
The target branch should be updated to `main` before merging. Wait for the current base branch to be merged.

**Review by parthea — APPROVED** [review]

File: gapic/templates/%namespace/%name_%version/%sub/services/%service/_shared_macros.j2
Comment (vchudnov-g) [inline gapic/templates/%namespace/%name_%version/%sub/services/%service/_shared_macros.j2:156]: We really shouldn't put links to internal issues in open-source code, but since this is meant to be temporary, it's alright...though I would annotate a bit further:

```suggestion
    {# TODO (ohmayr): Make this unconditional once REST interceptors are supported for async. Googlers, see internal tracking issue: b/362949568 #}
```

**Review by vchudnov-g — APPROVED** [review]

File: gapic/templates/%namespace/%name_%version/%sub/services/%service/_shared_macros.j2
Comment (ohmayr) [reply gapic/templates/%namespace/%name_%version/%sub/services/%service/_shared_macros.j2]: addressed.

**Pull request merged**
